### PR TITLE
bugfix: revise search algorithm for GitHub token

### LINKED ISSUE #156: Unit test for GitHub token fails when specific file name is provided
In #151, the code looks in more places for GitHub credentials.  Now, a unit test fails.

```
====================================================================== FAILURES ======================================================================
____________________________________________________________ test_GitHub_credentials_none ____________________________________________________________

    def test_GitHub_credentials_none():
        token = github_handler.get_GitHub_credentials("This file does not exist.",)
>       assert token is None
E       AssertionError: assert 'the_local_github_token_here' is None

punx/tests/test_github_handler.py:144: AssertionError
============================================================== short test summary info ===============================================================
FAILED punx/tests/test_github_handler.py::test_GitHub_credentials_none - AssertionError: assert 'the_local_github_token_here' is None
=========================================================== 1 failed, 85 passed in 28.60s ============================================================
```

In this case, `github_handler.get_GitHub_credentials()` returns the token since it is found in an alternative.  But the test specifies a file for the token and expects no token to be turned.

Either revise `github_handler.get_GitHub_credentials()` (to only look in the provided file if that is defined) or remove the unit test.

### PR DESCRIPTION
**Pull request by prjemian**
- FIX #156

### CODE REVIEW

**Comment by prjemian** [comment]
@carterbox This is ready for review.

File: punx/github_handler.py [hunk 1]
```diff
@@ -81,20 +95,32 @@ def get_GitHub_credentials(creds_file_name=""):
...
-
-    for name in [
-        creds_file_name,
-        os.path.join(os.path.dirname(__file__), CREDS_FILE_NAME),
-        os.path.join(os.path.expanduser("~"), ".config", CREDS_FILE_NAME),
-        os.path.join(os.path.expanduser("~"), ".config", "punx", CREDS_FILE_NAME),
-    ]:
-        if os.path.exists(name):
-            return open(name, "r").read().strip().split()[-1]
+    home = pathlib.Path.home()
```
Comment (carterbox) [inline punx/github_handler.py:100]: This line will raise an uncaught error if the home cannot be resolved, but the docs say that this function should return a token from environment variables regardless of whether there is a home directory.

https://docs.python.org/3/library/pathlib.html#pathlib.Path.home

**Review by carterbox — CHANGES_REQUESTED** [review]
Wow, this function has gotten pretty complicated! 😆

What if the search order is in order of user-controlled to developer-controlled because this function is not part of the public API? Or the opposite ordering?

^ more user-y
`GH_TOKEN`
`GITHUB_TOKEN`
``~/.config/__github_creds__.txt``
``~/.config/punx/__github_creds__.txt``
``creds_file_name`` This is the optional parameter; it is a path to any arbitrarily named text tile.
``[punx source directory]/__github_creds__.txt``
v more developer-y

**Comment by prjemian** [comment]
I agree this is too complicated.  Can we simplify to require _only_ either of the two environment variables?

**Comment by carterbox** [comment]
Do you mean the search order would be:
```
GH_TOKEN
GITHUB_TOKEN
```
and the function would take no parameters? Sounds good to me.

**Comment by prjemian** [comment]
Yes

**Comment by prjemian** [comment]
Now, still returns `None` if token not found but also issues a warning (that appears in the CI workflow):

```python
================================================================== warnings summary ==================================================================
punx/tests/test_github_handler.py::test_connect_repo
punx/tests/test_github_handler.py::test_connected_GitHub_Repository_Reference
punx/tests/test_github_handler.py::test_Github_download_default
  /home/prjemian/Documents/projects/prjemian/punx/punx/github_handler.py:80: UserWarning: Did not find environment variables GH_TOKEN or GITHUB_TOKEN
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================================================== 8 passed, 3 warnings in 0.87s ============================================================
```

**Comment by prjemian** [comment]
@carterbox - Docs will be modified if you agree on this approach.

File: punx/github_handler.py [hunk 2]
```diff
@@ -57,44 +58,38 @@
...
-            return open(name, "r").read().strip().split()[-1]
+        token = os.environ.get(variable)
+        if token is not None:
+            return token
+
+    warnings.warn(
+        UserWarning(
+            "Did not find environment variables GH_TOKEN or GITHUB_TOKEN"
+        )
+    )
```
Comment (carterbox) [inline punx/github_handler.py:84]:
```suggestion
    warnings.warn(
        "Did not find environment variables GH_TOKEN or GITHUB_TOKEN",
        UserWarning,
    )
```
https://docs.python.org/3/library/warnings.html#warnings.warn

File: punx/github_handler.py [hunk 3]
```diff
@@ -57,44 +58,38 @@
...
+    # TODO: move next conent to documentation.
+    # warnings.warn(
+    #     "Did not find environment variables GH_TOKEN or GITHUB_TOKEN"
+    #     " which provide the GitHub API token necessary to download"
+    #     " resources from GitHub through its API.  You may experience"
+    #     " restrictions on the amount of content that can be downloaded"
+    #     " over a chort intervale (such as an hour or so)."
+    # )
```
Comment (carterbox) [inline punx/github_handler.py:92]: I love a helpful warning/error message like this one. Why do you want to hide it in the docs?

**Review by carterbox — COMMENTED** [review]
I agree with the approach overall. Comments about warning implementation below.

File: punx/github_handler.py (hunk 3 above)
Comment (prjemian) [reply punx/github_handler.py:92]: Thought it was too verbose.  After editing the `chort intervale` typo, I'll re-enable this one.

File: punx/github_handler.py [hunk 4]
```diff
@@ -57,44 +58,33 @@
...
-        if os.path.exists(name):
-            return open(name, "r").read().strip().split()[-1]
+        token = os.environ.get(variable)
+        if token is not None:
+            return token
+
+    warnings.warn(
+        "Did not find environment variables GH_TOKEN or GITHUB_TOKEN"
+        " which provide the GitHub API token necessary to download"
+        " resources from GitHub through its API.  You may experience"
```
Comment (carterbox) [inline punx/github_handler.py:83]:
```suggestion
        " resources from GitHub through its API. You may experience"
```

**Review by carterbox — APPROVED** [review]

File: punx/github_handler.py (hunk 4 above)
Comment (prjemian) [reply punx/github_handler.py:83]: This is a matter of style.  I want to leave it as two spaces to be consistent with all my other writing.

**Comment by prjemian** [comment]
@carterbox Thanks for the review.  Feel free to merge and delete the branch.

**Pull request merged**